script.js

```diff
@@ -1 +1,406 @@
-// Please implement exercise logic here
+/* ##############
+## GLOBAL VARIABLES ##
+############## */
+// boardSize has to be an even number
+const boardSize = 4;
+let board = [];
+let boardEl;
+let boardElement;
+const gameInfo = document.createElement("div");
+let firstCard = null;
+let firstCardElement;
+let cardElement;
+let deck;
+let square;
+let squareDeco;
+let overlay;
+let userName;
+let gridContainer;
+let scoreElement;
+let buttonWrapper;
+let goStart;
+let goReset;
+let timeRemaining;
+let minute = 3;
+let sec = 0;
+let score = 0;
+let lockBoard = true; // use flag to not let user click while waiting for timeout and start game
+let handle;
+
+/* ##############
+## HELPER FUNCTION ##
+############## */
+const makeDeck = () => {
+  let newDeck = [];
+
+  // Initialise an array of the 4 suit in our deck. We will loop over this array.
+  const suit = [
```

Comment:
Arrays are better named as plurals, and in the for loop it’s more clear to write something like `for suit in suits` 😉